### ISSUE TITLE
False positive when decorating a method with the same name as another method using PEP 695 type parameter syntax

### ISSUE THREAD

**Issue by luwqz1** [microsoft/pyright#9292]
I found this example to repro:

```python
import typing


def foo(func: typing.Callable[..., typing.Any]) -> typing.Callable[..., typing.Any]: ...


class Bar:
    @foo  # "foo" is unbound
    def bar(self) -> None: ...

    def foo[T](self, value: T) -> T: ...
```

[pyright playground](https://pyright-play.net/?pythonVersion=3.13&strict=true&code=JYWwDg9gTgLgBDAnmYA7A5gWAFA5wEwFMAzOYiCACmIFdUBjALgWTXQDoBhAQwBtfuAI16EA2uwkAaFigzsAgqkQBdAJRwAtAD4ZbLnwHCxE9tKSyOilcxN5c2egIDOTuACFuURjji%2B4AAXIIHz8iUkFPSidCXmJ1bTgAOQhUQhsJOz84MLIKUQAVZSiY4mkANz4aNLh8%2BJ189PYcIA)

**Comment by erictraut** [comment]
I assume that you mean a "false positive" rather than "false negative" because pyright is reporting an error when it shouldn't be.

This issue is the same underlying cause as #9280, which is fixed in the main branch. It will be addressed in the next release.

**Comment by luwqz1** [comment]
> I assume that you mean a "false positive" rather than "false negative" because pyright is reporting an error when it shouldn't be.

Oops, sorry, thanks for the remark, i sometimes confuse "false positive" and "false negative," even though i know what they mean

> This issue is the same underlying cause as #9280, which is fixed in the main branch. It will be addressed in the next release.

I assumed that it might be the same issue, but i decided to report it anyway. Thank you so much for your response, im glad you were able to fix this bug so quickly yesterday!

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.386
